File: c/easy/minimum_difference_between_highest_and_lowest_of_k_scores.c
#include <stdlib.h>
#include <limits.h>

int compare(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

int minimumDifference(int* nums, int numsSize, int k) {
    if (k == 1) return 0;

    qsort(nums, numsSize, sizeof(int), compare);

    int min_difference = INT_MAX;

    for (int left = 0; left <= numsSize - k; left++) {
        int right = left + k - 1;
        int current_difference = nums[right] - nums[left];
        if (current_difference < min_difference) {
            min_difference = current_difference;
        }
    }

    return min_difference;
}

File: rust/easy/minimum_difference_between_highest_and_lowest_of_k_scores.rs
pub fn minimum_difference(mut nums: Vec<i32>, k: usize) -> i32 {
    if k == 1 {
        return 0;
    }

    let k = k as usize;

    nums.sort();

    let mut min_difference = i32::MAX;

    for left in 0..=nums.len() - k {
        let right = left + k - 1;
        let current_difference = nums[right] - nums[left];
        min_difference = min_difference.min(current_difference);
    }

    min_difference
}
